OpenSim/Simulation/Model/PhysicalFrame.cpp

```diff
@@ -237,22 +238,37 @@ void PhysicalFrame::convertDisplayGeometryToGeometryXML(SimTK::Xml::Element& bod
                 geomFile = geomFileIter->getValueAs<SimTK::String>();
             }
             // transform
-            SimTK::Vec6 localXform(0.);
-            SimTK::Xml::element_iterator localXformIter = displayGeomIter->element_begin("transform");
-            if (localXformIter != displayGeomIter->element_end()) {
-                localXform = localXformIter->getValueAs<SimTK::Vec6>();
+            SimTK::Vec6 innerXformVec6(0.);
+            SimTK::Xml::element_iterator innerXformIter = displayGeomIter->element_begin("transform");
+            if (innerXformIter != displayGeomIter->element_end()) {
+                innerXformVec6 = innerXformIter->getValueAs<SimTK::Vec6>();
             }
-            if (localXform.norm() > SimTK::Eps) {
-                // Create a Frame
+            // Old geometry/visibleObject could specify 2 transforms one per DisplayGeometry 
+            // and one for the whole visibleObject.
+            // We'll attach to the frame only if both transforms are identity
+            attachToThisFrame = (innerXformVec6.norm() < SimTK::Eps)
+                && (outerTransformVec6.norm() < SimTK::Eps);
+            // In practice no models were found where both innerTransform and outerTransform were
+            // non trivial or scale factors were specified in both
+            if (!attachToThisFrame) {
+                // Create a Vec6 by composing inner and outer transforms
+                SimTK::Transform outerTransform(outerTransformVec6.getSubVec<3>(3));
+                outerTransform.updR().setRotationToBodyFixedXYZ(outerTransformVec6.getSubVec<3>(0));
+                SimTK::Transform innerTransform(innerXformVec6.getSubVec<3>(3));
+                innerTransform.updR().setRotationToBodyFixedXYZ(innerXformVec6.getSubVec<3>(0));
+                SimTK::Transform composed = outerTransform.compose(innerTransform);
```

Comment:
Thanks @aseth1 for the review. The two models I tested with that failed to visualize properly had transforms in exactly one of the two places. One of them was added as a test case (double_pendulum) the other tested locally (basically Carmichael's model that I didn't want to add to the repo). We can manufacture models that have both non-trivial transforms but to go thru this only to collapse into one-transform doesn't seem quite justified. Considering how rare that may have happened (if ever) I'd be happy to throw an exception instead if both transforms are non-trivial.